packages/analytics/dashboard-renderer/src/composables/useExternalLinkCreator.ts

```diff
@@ -0,0 +1,12 @@
+import { useExternalLinkCreator } from '@kong-ui-public/entities-shared'
+
+export default (routeParams: string[]): string => {
+  // TODO: may have to pass in geo from consuming app
+  const activeGeoCode = 'us'
+
+  const parsedRouteParams = routeParams.map(rp => rp.replace('{geo}', activeGeoCode))
+
+  // TODO: replace `{idTuple0}` and `{idTuple1}` with actual record ID vales
+
+  return useExternalLinkCreator(parsedRouteParams)
+}
```

Comment:
if this isn't going to be shared amongst other charts in the dashboard, the logic could all be stashed inside `generateEntityUrl` in TopNTableRenderer most likely.

Reply:
We don't need use external link creator here 

The host app is supposed to do this, the host app supplies the output of external link creator to the dashboard and the renderer  replaces the appropriate ids